data/src/strapi.py

```diff
@@ -0,0 +1,93 @@
+from typing import Union
+from requests import Session
+from getpass import getpass
+from pathlib import Path
+from functools import cached_property
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Strapi:
+    session: Union[str, None] = None
+    url: Union[str, None] = None
+
+    def __init__(self, url):
+        self.url = url
+        self.session = Session()
+
+    def login(
+        self,
+        username: Union[str, None] = None,
+        password: Union[str, None] = None,
+        jwt: Union[str, None] = None,
+    ):
+        """Login to Strapi and set the JWT token in the session headers
+        Remember that strapi has 3 types of JWT tokens:
+        - API tokens: for public access
+        - Authenticated user tokens: for authenticated users
+        - Admin tokens: for admin users
+
+        only authenticated user tokens are valid for the content manager get Collections.
+        """
+        if not jwt:
+            if not username:
+                username = getpass("Enter your username: ")
+            if not password:
+                password = getpass("Enter your password: ")
+
+            try:
+                r = self.session.post(
+                    f"{self.url}/api/auth/local",
+                    json={"identifier": username, "password": password},
+                    headers={"accept": "application/json"},
+                )
+                r.raise_for_status()
+                jwt = r.json().get("jwt")
+            except Exception as e:
+                logger.error(r.json())
+                raise ValueError(r.json().get("error"))
+
+        self.session.headers.update({"Authorization": f"Bearer {jwt}"})
+
+        return self
+
+    def logout(self) -> None:
+        self.session.close()
+
+    @cached_property
+    def getCollections(self):
+        response = self.session.get(f"{self.url}/content-manager/collection-types")
```

Comment:
sorry, cannot get it to work, maybe you can help :) as far as I can tell, there is no such endpoint `/content-manager/collection-types` (strapi routes:list); there is `/content-manager/collection-types/:model`. There is an endpoint `/content-manager/content-types` which I think would work here, possibly with additional filtering.